### ISSUE TITLE
Select Widgets don't get localized

### ISSUE THREAD

**Issue by taleteller** [contao-bootstrap/form#10]
I am not entirely sure if this bug belongs to contao-bootstram/form or [netzmacht/contao-form-helper](https://github.com/netzmacht/contao-form-helper) since it touces both of them. Form brings the asset while helper handles their import.

However, when adding a select field into the form of a foreign localized webpage, you still get english text within the select field. I am not talking about the options you load into it, but control text. For example in the empty select box of a german webpage, you will read "Nothing selected' instead of 'Bitte wählen ...' as its supposed to be. 

This happens because the (sadly undocumented) localization of the bootstrap-select asset does not get handled at all. But they are bundled with. 

To hotfix the problem, i modified the bootstrap/select.html5 template as followes: 

```<?php
\Netzmacht\Contao\FormHelper\Helper::generate($this);
$selectLocalizations = array('cz'=>'cz_CZ', 'de'=>'de_DE', 'cl'=>'es_CL', 'eu'=>'eu', 'fr'=>'fr_FR', 'it'=>'it_IT', 'nl'=>'nl_NL', 'pl'=>'pl_PL', 'br'=>'pt_BR', 'ru'=>'ru_RU', 'cn'=>'zh_CN');
if ($selectLocalizations[$GLOBALS['TL_LANGUAGE']]) {
		$GLOBALS['TL_JAVASCRIPT'][] = 'system/modules/bootstrap-form/assets/bootstrap-select/i18n/defaults-'.$selectLocalizations[$GLOBALS['TL_LANGUAGE']].'.min.js';
}
```

This change maps the internal contao language code to the coresponding asset and adds it to the loading list. English is omitted, since its hardcoded. I dont think this is the 'correct' solution for a permanent fix, because asset loading is handled in 'contao-bootstrap.php'. However I am not even sure if there is a way to conditional load them there as required. Another way would be to add an option, but it would change the configuration of a default element. 

PS: My bootstrap-datepicker form field is done, i will publish it when i got time to tidy it up a little.

**Comment by dmolineus** [comment]
You are at the right repository. The styled selects are only provided by the bootstrap form component.

Nice you're figured it out. At the moment localization of the select is simply not supported. But I like your solution.

What will happen if I have a langauge like `de-DE` defined? Will `$GLOBALS['TL_LANGUAGE']` be just `de` or is an extra mapping of language accents required?

**Comment by Taleteller** [comment]
From what I know Contao uses ISO 639-1 codes for language definitions, as is the lang attribute from the html tag. Using a different declaration like 'de-DE' would get you problems, for instance I am not even sure browser language recognition would work reliable. 

However, using `$selectLocalizations[strtolower(substr($GLOBALS['TL_LANGUAGE'],-2))]` would be more tolerant at the cost of some speed.

**Comment by dmolineus** [comment]
I've added it in 1ce57ad1b4d73a656b298b8888e3dfa2958ce77d (develop branch). Would be great if you could test it.

Wanna release 1.0.0 this week. So I'm looking forward to your feedback.

**Comment by taleteller** [comment]
Yes that solution works for me as well. However, the asset [silviomoreto/bootstrap-select](https://github.com/silviomoreto/bootstrap-select) had changes in the past half year, and should be updated as well. For example the french localization file is missing in your checkout. 

Besides my [initial datepicker release](https://github.com/taleteller/bootstrap-datepicker) is available at the packagist under 'taleteller/bootstrap-datepicker'. Might not be first class code, but works ;)

**Comment by dmolineus** [comment]
> Yes that solution works for me as well. However, the asset silviomoreto/bootstrap-select had changes in the past half year, and should be updated as well. For example the french localization file is missing in your checkout.

I've updated it yesterday, in 0e33f641b6387df09c85681f3a2c5f407ad05e9f. ;-)

> Besides my initial datepicker release is available at the packagist under 'taleteller/bootstrap-datepicker'. Might not be first class code, but works ;)

Great, gonna have a look on it.

**Comment by dmolineus** [comment]
Released as 1.0.0.
